src/public/index.php

```diff
@@ -0,0 +1,14 @@
+<?php
+  require_once(dirname(__DIR__, 1)."/handler/PostHandler.php");
+
+  $request_url = $_SERVER["REQUEST_URI"];
+  $request_method = $_SERVER["REQUEST_METHOD"];
+
+  if ($request_method === "GET" && $request_url === "/") {
+    PostHandler::getPostListPage();
+    
+  } else if ($request_method === "POST" && $request_url === "/post")  {
+    PostHandler::createPost();
+
+  } else if ($request_method === "GET" && $request_url === "/post/:id"){
```

Comment:
```suggestion
  if ($request_method === "GET" && $request_url === "/") {
    PostHandler::getPostListPage();
    
  } else if ($request_method === "POST" && $request_url === "/posts")  {
    PostHandler::createPost();

  } else if ($request_method === "GET" && $request_url === "/posts/:id"){
```

RESTの設計的にエンドポイントのリソース名は複数形が良いと思います。
https://qiita.com/NagaokaKenichi/items/6298eb8960570c7ad2e9#1%E8%A4%87%E6%95%B0%E5%BD%A2%E3%81%AE%E5%90%8D%E8%A9%9E%E3%82%92%E5%88%A9%E7%94%A8%E3%81%99%E3%82%8B